### PR TITLE
Add a feature to display error line number

### LINKED ISSUE #240: Incorrect error line number for the editor
For example, it's line 9, not 382 for the following app.  

image

### PR DESCRIPTION
**Pull request by algomaster99**
Fixes #240 

### Fix
These changes add a feature which will also display the number of line where the error exists.

### Demo
Screenshot from 2019-03-09 19-35-32

### Advantage
User can find the error more easily since the line number can be located.

### CODE REVIEW

**Comment by algomaster99** [comment]
@domoritz Can you please review my pull request and suggest any changes required?

**Comment by domoritz** [comment]
Sweet, thank you! 

Before I merge this, could you check whether the monaco editor has a way to access the error it is producing? Maybe we can just hook into that. Thanks!

**Comment by algomaster99** [comment]
@domoritz I couldn't find a way to extract line number from the monaco editor. Although I found a [function](https://microsoft.github.io/monaco-editor/api/modules/monaco.editor.html#getmodelmarkers) in their API which could have helped.
But this function is not defined for the react integration. That's all, I could conclude. Any suggestions you have which can solve this issue?

File: src/reducers/index.ts [hunk 1]
```diff
@@ -47,11 +47,28 @@ function parseVega(state: State, action: SetVegaExample | UpdateVegaSpec | SetGi
       vegaSpec: spec,
     };
   } catch (e) {
+    const code = action.spec;
+    const pattern = /\d+/;
```
Comment (algomaster99) [inline src/reducers/index.ts]: #### Before
```
const pattern = /(end|\d+)/
```
#### Now
```
const pattern = /\d+/
```
I only needed to match numbers. This change required me to force-push the branch.

**Comment by domoritz** [comment]
> But this function is not defined for the react integration.

That could be a bug in the typings. I have publishing permissions for https://github.com/superRaytin/react-monaco-editor so you can send a PR to fix it.

**Comment by domoritz** [comment]
See https://github.com/Microsoft/monaco-editor/issues/30 for example.

**Comment by algomaster99** [comment]
@domoritz 
> That could be a bug in the typings.

This function is anyway not mentioned in the `README.md` so I don't think if it requires a pull request. Although `this.refs.monaco.editor` did throw an error: `TS2339: Property 'editor' does not exist on type 'ReactInstance'. Property 'editor' does not exist on type 'Component<any, {}, any>'`. I'll see to it and submit a PR.

**Comment by domoritz** [comment]
Looks like you can't access the editor property. You can ignore the type error and see whether it works first.

**Comment by algomaster99** [comment]
@domoritz I can't seem to find a way to access errors using `react-monaco-editor`.

**Comment by domoritz** [comment]
What about `monaco-editor`?

**Comment by algomaster99** [comment]
@domoritz Haven't checked that out yet. That's a major step to bring in though. I'll set-up a dummy project and try it out first then let you know what will be better. :smile:

**Comment by domoritz** [comment]
Well, you can access the monaco editor directly in the vega editor. You might have to ignore the typings.

**Comment by algomaster99** [comment]
```
const model = this.refs.editor['editor'];
model.getModelMarkers({});
```
Source: https://stackoverflow.com/questions/43463344/getting-errors-from-monaco-editor

This is how I tried solving this issue but it throws an error that `getModelMarkers` isn't a function.

Although
```
model.getModel()
```
works and according to [docs](https://microsoft.github.io/monaco-editor/api/modules/monaco.editor.html#getmodels), `getModel()` and `getModelMarkers()` are in the same class. Line 858 and 867.
Class file: https://github.com/Microsoft/monaco-editor/blob/ddbd7148e4afd00c60e4262f3dc2fb5f1ce18fe2/monaco.d.ts#L858

**Comment by domoritz** [comment]
We're they added in a particular version?

**Comment by algomaster99** [comment]
The `monaco-editor` version we are using and this api's version is same i.e. 0.16.0

**Comment by domoritz** [comment]
Not sure but maybe the last comment in https://github.com/Microsoft/monaco-editor/issues/906 helps. Can you grab the editor from somewhere else other than the onload callback?

**Comment by algomaster99** [comment]
@domoritz Yes, I can get it from anywhere after the editor has mounted

**Comment by domoritz** [comment]
Does it have getModelMarkers then as described in the issue I linked to?

**Comment by domoritz** [comment]
Okay, let's look into this later then.

**Comment by algomaster99** [comment]
@domoritz I didn't add this feature to vega-lite editor. I'll add it and submit a PR soon.

**Comment by domoritz** [comment]
Oh, I missed that. Thanks!

**Comment by yijiao19** [comment]
I'm wondering how to use this feature in the monaco-editor?
I cannot find any description in the API page [https://microsoft.github.io/monaco-editor/api/index.html](url)

**Pull request merged**